utils/cliutils/utils.go

```diff
@@ -809,3 +809,15 @@ func getDebFlag(c *cli.Context) (deb string, err error) {
 	}
 	return deb, nil
 }
+
+// ExtractBoolFlagFromArgs Extracts a boolean flag from the args and removes it from the slice.
+func ExtractBoolFlagFromArgs(filteredArgs *[]string, flagName string) (value bool, err error) {
```

Comment:
Let's add a unit test for this function.
You can use the below AI generated test as a base to start from - 
```
func TestExtractBoolFlagFromArgs(t *testing.T) {
	testCases := []struct {
		name          string
		args          []string
		flagName      string
		expectedValue bool
		expectedErr   bool
		expectedArgs  []string
	}{
		{
			name:          "Flag present as --flagName (implied true)",
			args:          []string{"somecmd", "--flagName", "otherarg"},
			flagName:      "flagName",
			expectedValue: true,
			expectedErr:   false,
			expectedArgs:  []string{"somecmd", "otherarg"},
		},
		{
			name:          "Flag present as --flagName=true",
			args:          []string{"somecmd", "--flagName=true", "otherarg"},
			flagName:      "flagName",
			expectedValue: true,
			expectedErr:   false,
			expectedArgs:  []string{"somecmd", "otherarg"},
		},
		{
			name:          "Flag present as --flagName=false",
			args:          []string{"somecmd", "--flagName=false", "otherarg"},
			flagName:      "flagName",
			expectedValue: false,
			expectedErr:   false,
			expectedArgs:  []string{"somecmd", "otherarg"},
		},
		{
			name:          "Flag not present",
			args:          []string{"somecmd", "otherarg"},
			flagName:      "flagName",
			expectedValue: false,
			expectedErr:   false,
			expectedArgs:  []string{"somecmd", "otherarg"},
		},
		{
			name:          "Flag present with invalid value",
			args:          []string{"somecmd", "--flagName=invalid", "otherarg"},
			flagName:      "flagName",
			expectedValue: false,
			expectedErr:   true,
			expectedArgs:  []string{"somecmd", "otherarg"},
		},
		{
			name:          "Flag present as -flagName (should not be found)",
			args:          []string{"somecmd", "-flagName", "otherarg"},
			flagName:      "flagName",
			expectedValue: false,
			expectedErr:   false,
			expectedArgs:  []string{"somecmd", "-flagName", "otherarg"},
		},
		{
			name:          "Flag present multiple times",
			args:          []string{"somecmd", "--flagName", "--flagName=false", "otherarg"},
			flagName:      "flagName",
			expectedValue: false, // Last occurrence takes precedence
			expectedErr:   false,
			expectedArgs:  []string{"somecmd", "otherarg"},
		},
	}

	for _, tc := range testCases {
		t.Run(tc.name, func(t *testing.T) {
			argsCopy := append([]string(nil), tc.args...) // Make a copy to avoid modifying the original
			value, err := ExtractBoolFlagFromArgs(&argsCopy, tc.flagName)

			if tc.expectedErr {
				assert.Error(t, err)
			} else {
				assert.NoError(t, err)
			}

			assert.Equal(t, tc.expectedValue, value)
			assert.Equal(t, tc.expectedArgs, argsCopy)
		})
	}
}

```